sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/types/DataTypeUtils.scala

```diff
@@ -64,6 +64,8 @@ object DataTypeUtils {
    * - Both types are structs and have the same number of fields. The type and nullability of each
    *   field from read/write is compatible. If byName is true, the name of each field from
    *   read/write needs to be the same.
+   * - it is user defined type and its underlying sql type is same as the read type.
+   * - the read type is user defined type and its underlying sql type is same as the write type.
```

Comment:
Ditto. please remove `-` in the continuation of the paragraph.

Reply:
Okay